GTMAppAuthSwift/Sources/KeychainStore.swift

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import Security
+// Ensure that we import the correct dependency for both SPM and CocoaPods since
+// the latter doesn't define separate Clang modules for subspecs
+#if SWIFT_PACKAGE
+import AppAuthCore
+import GTMSessionFetcherCore
+#else
+import AppAuth
+import GTMSessionFetcher
+#endif
+
+/// A helper providing a concrete implementation for saving and loading auth data to the keychain.
+@objc(GTMKeychainStore)
+public final class KeychainStore: NSObject {
+  private var keychainHelper: KeychainHelper
+  // Needed for `AuthStateStore` and listed here because extensions cannot add stored properties
+  @objc public var itemName: String
+
+  /// An initializer for testing to create an instance of this keychain wrapper with a given helper.
+  ///
+  /// - Parameters:
+  ///   - itemName: The `String` name for the credential to store in the keychain.
+  ///   - keychainHelper: An instance conforming to `KeychainHelper`.
+  init(itemName: String, keychainHelper: KeychainHelper) {
```

Comment:
Will we be providing dataProtectionKeychain and keychainAccessGroup configurability via the keychainHelper parameter here?

Reply:
https://github.com/google/GTMAppAuth/pull/201 does this.